configuration/uart.md

```diff
@@ -21,31 +17,33 @@ Broadcom gives each UART a name and number in their SoC documentation, and this
 | BCM2837 | PL011 | mini UART |
 | BCM2711 | PL011 | mini UART | PL011 | PL011 | PL011 | PL011 |
 
- See the [Raspberry Pi 4 Model B Datasheet](https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2711/rpi_DATA_2711_1p0_preliminary.pdf) section _5.1.2 GPIO Alternate Functions_, for details on exactly which UART can appear on which GPIO numbers. Earlier models are similar, but have only the first two UARTs. Take care to line up the numbers: TXD0 and RXD0 always belong to UART0, TXD1 to UART1 and so on. Also, remember that the GPIO numbering is on the CPU, and is different than the GPIO pin numbers on the 40-pin J8 header.
+Inside the Raspberry Pi, the UARTs are able to connect in several ways. UART0 and UART1 can connect to either the GPIO lines, or the onboard Bluetooth module, if there is one. UARTs two through five connect to the GPIO lines and nowhere else.
 
-Inside the Raspberry Pi, the UARTs are able to connect in several ways. UART0 and UART1 can connect to either the GPIO lines or the onboard Bluetooth module, if there is one. UARTs two through five connect to the GPIO lines and nowhere else.
+See the [Raspberry Pi 4 Model B Datasheet](https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2711/rpi_DATA_2711_1p0_preliminary.pdf) section _5.1.2 GPIO Alternate Functions_, for pinout details. Earlier models are similar but with only the first two UARTs.
 
 ### Voltages
 
-All the onboard Raspberry Pi UARTs are 3.3V [logic level](https://en.wikipedia.org/wiki/Logic_level) only, sometimes called ['CMOS'](https://en.wikipedia.org/wiki/CMOS) level. Do not connect Raspberry Pi UARTs to 5V circuits, sometimes called ['TTL'](https://en.wikipedia.org/wiki/Transistor%E2%80%93transistor_logic) level, because the Raspberry Pi will be damaged. [RS-232](https://en.wikipedia.org/wiki/RS-232) uses +/- 12V signals, which are right out.
-
-There are chips called ['level shifters'](https://en.wikipedia.org/wiki/Level_shifter) which convert voltages between levels. You can find them either as individual ICs for using in your own designs, or integrated into other devices to make them compatible. They're inexpensive and readily available from third parties.
+All the onboard Raspberry Pi UARTs are 3.3V ([CMOS](https://en.wikipedia.org/wiki/CMOS) [logic level](https://en.wikipedia.org/wiki/Logic_level)). Do not connect Raspberry Pi UARTs to 5V ([TTL logic level](https://en.wikipedia.org/wiki/Transistor%E2%80%93transistor_logic)) circuits, because the Raspberry Pi will be damaged. [Level shifters](https://en.wikipedia.org/wiki/Level_shifter) are inexpensive and readily available.
```

Comment:
Maybe worth mentioning 3.3V USB UART adaptors too? (as I suspect they'll be a more common approach than using a level-shifter)

Is the distinction between CMOS and TTL (rather than just between 3.3V and 5V) necessary? :shrug: (I'm not a hardware engineer :wink: )

Reply:
I had included the names "TTL" and "CMOS" because product descriptions often express these names rather than the numerical voltage specs. An internet search for "USB serial port" shows quite a few products listed as "TTL" but none as "CMOS". Additionally, I had thought that TTL is pretty explicitly 5V logic levels, but many of the devices listed as TTL claim to be 3.3V and 5V compatible--if I read them correctly. I have to think they signal with 3.3V and hope it's close enough to be correctly interpreted by a 5V interface.

Considering the ambiguity of "TTL" and the absence of "CMOS" terminology, I think we're better off dropping those names and going with the numeric voltages.

This will be corrected in the next rev.

Reply:
> All the onboard Raspberry Pi UARTs are 3.3V ([CMOS](https://en.wikipedia.org/wiki/CMOS) [logic level](https://en.wikipedia.org/wiki/Logic_level)). Do not connect Raspberry Pi UARTs to 5V ([TTL logic level](https://en.wikipedia.org/wiki/Transistor%E2%80%93transistor_logic)) circuits

The TTL connectors I've dealt with in the past use 3.3V logic, not 5V.